### DISCUSSION TITLE
How do you decide if something is a property of a Type-class vs an Individual-class?

### DISCUSSION THREAD

**Discussion by ManuelSimoes** [nemo-ufes/gufo#29] [Q&A] [answered]
For example: given that some weapon types are banned by law, it seems you could model this as a property of WeaponType or as a specialization (ForbiddenWeapon/AllowedWeapon) of Weapon. A law would typically ban a weapon type, not an individual weapon, so modelling the ban as a property of WeaponType seems to make more sense. But what exactly would go wrong if you model it as a specialization of Weapon? 
![image](https://github.com/nemo-ufes/gufo/assets/174717727/acbfde81-7dfe-47a3-b36a-efea96b29f30)

**Comment by jpalmeida** [comment, marked as answer]
In principle, both solutions are fine (I would add that "WeaponType" gufo:categorizes "Weapon" in the example in the lefthand side). Having a second-order type in your ontology such as "WeaponType" allows you to use it in the domain and range of properties. For example, you could have "WeaponType" as the range of object properties. 

Imagine that you also have a class called "Law". You could have an object property called "forbids" with domain "Law" and range "WeaponType".  

Or you could have an object property called "wasInventedBy" with domain "WeaponType" and range "Person". 

These object properties are not viable unless you have "WeaponType" in your ontology.
